smartclient-core/src/test/java/org/srg/smartclient/JDBCHandlerAddTest.java

```diff
@@ -0,0 +1,123 @@
+import org.srg.smartclient.utils.ContextualRuntimeException;
+import org.srg.smartclient.utils.Serde;
+
+import java.io.StringWriter;
+
+public class JDBCHandlerAddTest extends AbstractJDBCHandlerTest<JDBCHandler> {
+
+    @Override
+    protected Class<JDBCHandler> getHandlerClass() {
+        return JDBCHandler.class;
+    }
+
+    @Test
+    public void simpleAdd() throws Exception {
+
+        final DSRequest request = JsonTestSupport.fromJSON(new TypeReference<>(){}, """
+               {
+                 useStrictJSON : true,
+                 dataSource : "EmployeeDS",
+                 operationType : "ADD",
+                 componentId : "isc_ListGrid_5",
+                 startRow : 0,
+                 endRow : 0,
+                 textMatchStyle : "EXACT",
+                 data : {
+                   name: "A new Record"
+                 },
+                 oldValues : null
+               }                
+            """);
+
+        final DSResponse response;
+
+        try {
+            response = handler.handleAdd(request);
+        } catch (ContextualRuntimeException e) {
+            /*
+             * This exception handler can be used  to check and adjust
+             * ContextualRuntimeException.dumpContext_ifAny().
+             *
+             * Other than that it does not have any sense
+             */
+            final StringWriter sw = new StringWriter();
+            final ObjectMapper mapper = Serde.createMapper();
+
+            e.dumpContext_ifAny(sw, "  ", mapper.writerWithDefaultPrettyPrinter());
+            System.out.println( sw.toString());
+            throw new RuntimeException(e);
+        }
+        JsonTestSupport.assertJsonEquals("""
+                 {
+                     status: 0,
+                     data:[
+                        {
+                             id:1,
+                             name: 'admin'
+                         },
+                         {
+                             id:2,
+                             name: 'developer'
+                         },
+                         {
+                             id:3,
+                             name: 'UseR3'
+                         },
+                                                 {
+                             id:4,
+                             name: 'manager1'
+                         },
+                                                 {
+                             id:5,
+                             name: 'manager2'
+                         },
+                         {
+                             id:6,
+                             name: 'user2'
+                         },
+                         {
+                             id:7,
+                             name: 'A new Record'
+                         }
+                     ]
+                }""", response);
+    }
+
+    @Test
+    public void addingMustFails_if_idExists() throws Exception {
```

Comment:
addingMustFails_if_idExists should be addingMustFails_if_providedIdAlreadyExists